gtsfm/view_graph_estimator/view_graph_estimator_base.py

```diff
@@ -123,9 +145,9 @@ def compute_metrics(
         if len(two_view_reports) == 0:
             return GtsfmMetricsGroup(name="rotation_cycle_consistency_metrics", metrics=[])
 
-        input_edges = two_view_reports.keys()
+        valid_edges = self.__get_valid_input_edges(i2Ri1_dict)
```

Comment:
I recommend the following change: 
- call __get_valid_input_edges in create_computation_graph before calling run(). 
- remove invalid edges before calling run() so that derived classes can assume all edges are valid. 
- pass the same valid edges to run(), _filter_with_edges, and compute_metrics().

Reply:
happy to make this change now if youd like

Reply:
Thanks for taking a look Akshay. I would recommend we not modify the i2Ri1_dict container, in the spirit of only operating on the key tuples. Maybe calling get_valid_input_edges() beforehand is the best option, then, and passing this into each module.

Reply:
I do worry about cluttering the API with more input args, though. Maybe we should filter beforehand then.

Reply:
(to a new subset data structure)

Reply:
Please take a look at the most recent changes. Let me know if there are specific things you'd like to do differently, we can discuss offline before making more changes.

Reply:
i like it!!

Reply:
I'm "approving" my own PR. But let's wait for CI to shed some light on if things are truly working

Reply:
sounds good, Ill also wait for CI before approving.